codegenerator/cli/templates/static/codegen/src/eventFetching/chainWorkers/HyperSyncWorker.res

```diff
@@ -288,35 +289,52 @@ module Make = (
         | parsedEvents => parsedEvents
         }
 
-        pageUnsafe.items
-        ->Belt.Array.zip(parsedEvents)
-        ->Belt.Array.map(((item, event)): Types.eventBatchQueueItem => {
-          let {block, transaction, log: {logIndex}} = item
+        pageUnsafe.items->Belt.Array.mapWithIndex((index, item): Types.eventBatchQueueItem => {
+          let {block, transaction, log} = item
           let chainId = chain->ChainMap.Chain.toChainId
-          let (event, eventMod) = switch event
-          ->getNullableExn(~msg="Event was unexpectedly parsed as undefined", ~logger)
-          ->Converters.convertHyperSyncEvent(
-            ~eventModLookup,
+          let topic0 = log.topics->Js.Array2.unsafe_get(0)
+
+          let eventMod = switch eventModLookup->EventModLookup.get(
+            ~sighash=topic0,
             ~contractAddressMapping,
-            ~log=item.log,
-            ~block,
-            ~transaction,
-            ~chain,
+            ~contractAddress=log.address,
           ) {
-          | Ok(v) => v
-          | Error(exn) =>
-            let logger = Logging.createChildFrom(
-              ~logger,
-              ~params={"chainId": chainId, "blockNumber": block.number, "logIndex": logIndex},
-            )
-            exn->ErrorHandling.mkLogAndRaise(~msg="Failed to convert decoded event", ~logger)
+          | None => {
+              let logger = Logging.createChildFrom(
+                ~logger,
+                ~params={
+                  "chainId": chainId,
+                  "blockNumber": block.number,
+                  "logIndex": log.logIndex,
+                  "contractAddress": log.address,
+                  "topic0": topic0,
+                },
+              )
+              %raw(`null`)->ErrorHandling.mkLogAndRaise(
+                ~msg="Failed to lookup registered event",
```

Comment:
Improved error message (I'm not sure about null, but I don't have an exception here)

Reply:
I would suggest in these cases just create a new exception type that's searchable and could be pattern matched by rescript. But Either way think we can improve our error handling model